apps/content/services/asset_verse_text.py

```
@@ -0,0 +1,107 @@
+    """
+    latest = asset.get_latest_version()
+    # The key carries BOTH the selected row's identity and its last-modified
+    # stamp at FULL microsecond precision, so stale text can never outlive the
+    # version row that produced it:
+    # - a NEW AssetVersion row becomes latest -> different pk -> fresh key;
+    # - an IN-PLACE edit (portal PUT/PATCH re-saving the same row) bumps
+    #   updated_at (BaseModel auto_now) -> rotated key without any signals.
+    # Sub-second precision matters: consecutive saves routinely land within
+    # the same wall-clock second.
+    if latest is None:
+        version_token = "none"
```

Ruff (0.16.2)

[error] Possible hardcoded password assigned to: "version_token"

(S105)